src/modules/75-cf/components/CreateEnvironmentDialog/__tests__/EnvironmentDialog.test.tsx

```diff
@@ -72,4 +74,69 @@ describe('CreateEnvironmentButton', () => {
       expect(createEnvironmentButton).not.toBeDisabled()
     })
   })
+
+  describe('Create Environment Button', () => {
+    test('it should show appropriate error message if Environment name is empty', async () => {
+      jest
+        .spyOn(usePlanEnforcementMock, 'default')
+        .mockReturnValue({ isPlanEnforcementEnabled: false, isFreePlan: false })
+      jest.spyOn(useFeaturesMock, 'useFeature').mockReturnValue({ enabled: false })
+      renderComponent()
+
+      // open the modal
+      userEvent.click(screen.getByRole('button', { name: '+ environment' }))
+
+      // leave environemnt name blank and submit form
+      await act(async () => {
+        userEvent.click(screen.getByRole('button', { name: 'createSecretYAML.create' }))
+      })
+
+      expect(screen.getByText('fieldRequired')).toBeInTheDocument()
+    })
+
+    test('it should show appropriate error message if Environment name contains a non-ASCII character', async () => {
+      jest
+        .spyOn(usePlanEnforcementMock, 'default')
+        .mockReturnValue({ isPlanEnforcementEnabled: false, isFreePlan: false })
+      jest.spyOn(useFeaturesMock, 'useFeature').mockReturnValue({ enabled: false })
+
+      renderComponent()
+
+      // open the modal
+      userEvent.click(screen.getByRole('button', { name: '+ environment' }))
+
+      // enter invalid text and submit form
+      await act(async () => {
+        const environmentNameInputField = screen.getByRole('textbox', { name: '' })
+        userEvent.type(environmentNameInputField, 'à')
+
+        userEvent.click(screen.getByRole('button', { name: 'createSecretYAML.create' }))
+      })
+
+      expect(screen.getByText('common.validation.namePatternIsNotValid')).toBeInTheDocument()
```

Comment:
As above, no need to wrap in an `act` with `userEvent`. You may wanna wait for something to show before you try the action.

```suggestion
      await waitFor(() => expect(screen.getByRole('textbox', { name: '' })).toBeInTheDocument())
      
      await userEvent.type(screen.getByRole('textbox', { name: '' }), 'à')
      userEvent.click(screen.getByRole('button', { name: 'createSecretYAML.create' }))

     await waitFor(() => {
       expect(screen.getByText('common.validation.namePatternIsNotValid')).toBeInTheDocument()
     })
```

Reply:
I've put in a `waitFor` and it was okay. Does it still need the `await`?

Reply:
Yup, `waitFor` is asynchronous, so you need to `await` it